chain/chain/src/chain.rs

```diff
@@ -472,6 +472,10 @@ impl Chain {
                     self.transaction_validity_period,
                 );
 
+                if chain_update.check_header_known(header).is_err() {
```

Comment:
if there are other errors, this will swallow them. How about:
```
match chain_update.check_header_known(header) {
 Ok(_) => {},
 Err(e) => match e.kind() {
    ErrorKind::Unfit(_) => continue,
    _ => return e
 } 
};
```